prime-router/src/test/kotlin/fhirengine/engine/fhirRouterTests/FHIRBundleHelpersTests.kt

```diff
@@ -174,6 +213,28 @@ class FHIRBundleHelpersTests {
         assertThat(references).isNotEmpty()
     }
 
+    @Test
+    fun `test adding diagnosticreport references to bundle when no original provenance`() {
```

Comment:
If we do Jimmy's change, let's add a test similar to this but that makes sure that if the diagnostic report references already exist that they don't get doubled.

Reply:
Yes, the diagnostic reports will get duplicated currently. It does not check if it already exists, just blindly adds them, so I can create a tech debt ticket for this.